django/contrib/gis/gdal/datasource.py

```diff
@@ -89,12 +89,12 @@ def __getitem__(self, index):
         if isinstance(index, str):
             layer = capi.get_layer_by_name(self.ptr, force_bytes(index))
             if not layer:
-                raise IndexError('invalid OGR Layer name given: "%s"' % index)
+                raise IndexError('Invalid OGR Layer name given: %s.' % index)
```

Comment:
There was no test case for this and it looks as though this doesn't work as expected.

Test failure when attempting to add an assertion for an invalid name:

```
======================================================================
ERROR: test01_valid_shp (gis_tests.gdal_tests.test_ds.DataSourceTest)
Testing valid SHP Data Source files.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/lib64/python3.6/unittest/case.py", line 59, in testPartExecutor
    yield
  File "/usr/lib64/python3.6/unittest/case.py", line 605, in run
    testMethod()
  File ".../django/tests/gis_tests/gdal_tests/test_ds.py", line 91, in test01_valid_shp
    ds['invalid']
  File ".../django/django/contrib/gis/gdal/datasource.py", line 90, in __getitem__
    layer = capi.get_layer_by_name(self.ptr, force_bytes(index))
  File ".../django/django/contrib/gis/gdal/prototypes/errcheck.py", line 128, in check_pointer
    raise GDALException('Invalid pointer returned from "%s"' % func.__name__)
django.contrib.gis.gdal.error.GDALException: Invalid pointer returned from "OGR_DS_GetLayerByName"
```